common/src/lib.rs

```diff
@@ -76,7 +109,68 @@ pub mod ext {
             ext_ffi::new_uref(key_ptr);
             core::slice::from_raw_parts(key_ptr, UREF_SIZE)
         };
-        Key::from_bytes(slice).unwrap().0
+        deserialize(slice).unwrap()
+    }
+
+    fn fn_bytes_by_name(name: &String) -> Vec<u8> {
+        let (name_ptr, name_size, _bytes) = to_ptr(name);
+        let fn_size = unsafe { ext_ffi::function_size(name_ptr, name_size) };
+        let fn_ptr = alloc_bytes(fn_size);
+        unsafe {
+            ext_ffi::function_bytes(name_ptr, name_size, fn_ptr, fn_size);
+            Vec::from_raw_parts(fn_ptr, fn_size, fn_size)
+        }
+    }
+
+    pub fn fn_by_name(name: &String) -> Value {
+        let fn_bytes = fn_bytes_by_name(name);
+        Value::Contract(fn_bytes)
+    }
+
+    pub fn store_function(name: &String) -> Key {
+        let fn_bytes = fn_bytes_by_name(name);
+        let mut hasher = VarBlake2b::new(32).unwrap();
+        hasher.input(&fn_bytes);
+        let mut fn_hash = [0u8; 32];
+        hasher.variable_result(|hash| fn_hash.clone_from_slice(hash));
+        let key = Key::Hash(fn_hash);
+        let value = Value::Contract(fn_bytes);
+        write(&key, &value);
+        key
+    }
+
+    pub fn get_arg<T: BytesRepr>(i: u32) -> T {
+        let arg_size = unsafe { ext_ffi::size_of_arg(i) };
+        let dest_ptr = alloc_bytes(arg_size);
+        let arg_bytes = unsafe {
+            ext_ffi::get_arg(i, dest_ptr, arg_size);
+            core::slice::from_raw_parts(dest_ptr, arg_size)
+        };
+        deserialize(arg_bytes).unwrap()
```

Comment:
Can you add a `TODO` about handling properly the error case here? For example when a method expects `i32` but is passed something else.